### PR TITLE
Th-908: Bottom banner

### PR DESCRIPTION
**Pull request by nenonja**
## Description :sparkles:
Add bottom banner to front page. Also refactored topBanner to use Landingpage > topBanner fields.

## Issues :bug:
https://helsinkisolutionoffice.atlassian.net/browse/TH-908
## Testing :alembic:
start server locally and:
- Change bottom banner from cms: https://cms.test.kuva.hel.ninja/admin/pages/60/edit/ see how it changes on front page:  http://localhost:3000/fi/home 
- Try to change some other page: https://cms.test.kuva.hel.ninja/admin/pages/81/edit/ and preview the change:
http://localhost:3000/fi/home/81

## Additional notes :spiral_notepad:

### CODE REVIEW

File: src/domain/banner/bannerConstants.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const BANNER_SOME_IMAGE = '/images/activities_SoMe-share.jpg';
```
Comment (Sartonon) [inline src/domain/banner/bannerConstants.ts:1]: naming of this file is not in line with other files with constant values. Shoul we also name this "constants.ts"?

File: src/domain/landingPage/LandingPagePreview.tsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import React from 'react';
+import * as React from 'react';
```
Comment (Sartonon) [inline src/domain/landingPage/LandingPagePreview.tsx:1]: is there need to change this import style?

File: src/domain/banner/bannerUtils.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
+import { BannerPage, Maybe } from '../../generated/graphql';
```
Comment (Sartonon) [inline src/domain/banner/bannerUtils.ts:1]: naming of the file is not in line with the other constant files. Not saying this worse, might be better actually but should we change the naming of the files in all places?

File: src/domain/landingPage/__tests__/LandingPagePreview.test.tsx [hunk 4]
```diff
@@ -46,24 +56,35 @@ const renderComponent = () =>
...
     expect(
-      screen.getByRole('heading', { name: landingPageTitle })
+      screen.getByRole('heading', { name: topBannerTitle })
+    ).toBeInTheDocument();
+  });
+  await waitFor(() => {
+    expect(
+      screen.getByRole('heading', { name: bottomBannerTitle })
     ).toBeInTheDocument();
   });
```
Comment (Sartonon) [inline src/domain/landingPage/__tests__/LandingPagePreview.test.tsx:89]: `await screen.findByRole('heading', { name: bottomBannerTitle })`

would be better if we are waiting for an element

Comment (Sartonon) [reply src/domain/landingPage/__tests__/LandingPagePreview.test.tsx:89]: that avoids use of waitFor

File: src/util/mockDataUtils.ts [hunk 5]
```diff
@@ -241,28 +242,45 @@ export const fakeLandingPages = (
...
+    heroTopLayerImage: fakeLocalizedCmsImage(),
+    keywords: fakeLocalizedCmsKeywords(),
+    metaInformation: fakeLocalizedObject(),
+    pageTitle: fakeLocalizedObject(),
+    socialMediaImage: fakeLocalizedCmsImage(),
+    title: fakeLocalizedObject(),
+    titleAndDescriptionColor: fakeLocalizedObject('BLACK'),
+    __typename: 'LandingPage',
+    ...overrides,
+  };
```
Comment (Sartonon) [inline src/util/mockDataUtils.ts:264]: lets not get rid of merge function here. it merges objects deeploy and that is why it was used here

File: src/util/mockDataUtils.ts [hunk 6]
```diff
@@ -241,28 +242,45 @@ export const fakeLandingPages = (
...
+};
 
+export const fakeBanner = (overrides?: Partial<BannerPage>): BannerPage => {
+  return {
```
Comment (Sartonon) [inline src/util/mockDataUtils.ts:268]: here we could add merge

File: src/domain/banner/bannerUtils.ts (hunk 3 above)
Comment (nenonja) [reply src/domain/banner/bannerUtils.ts:1]: I would rather rename them all than have multiple files with the same name. At least in intelliJ it makes searching by filenames easier

File: src/util/mockDataUtils.ts (hunk 5 above)
Comment (nenonja) [reply src/util/mockDataUtils.ts:264]: mixed it up with lodash assign which can be replaced by es6 spread.

File: src/domain/banner/bannerUtils.ts (hunk 3 above)
Comment (Sartonon) [reply src/domain/banner/bannerUtils.ts:1]: vscode nicely shows the path for all the found files so having same names is not necessarily problem there but the case might be different for IntelliJ
Screenshot 2020-12-23 at 8 44 00

Comment (Sartonon) [reply src/domain/banner/bannerUtils.ts:1]: Screenshot 2020-12-23 at 8 46 39

another example

Comment (Sartonon) [reply src/domain/banner/bannerUtils.ts:1]: Screenshot 2020-12-23 at 8 47 27

and when file is open you can see the path

Comment (nenonja) [reply src/domain/banner/bannerUtils.ts:1]: IntelliJ have the same features but the problem is perhaps when you want to find all LandingPage related file:
Screenshot 2020-12-23 at 10 56 59 AM
You wont find the utils.ts
But now when you search banner releated file, bannerUtils is visible:
Screenshot 2020-12-23 at 10 57 13 AM

File: src/domain/landingPage/LandingPagePreview.tsx (hunk 2 above)
Comment (nenonja) [reply src/domain/landingPage/LandingPagePreview.tsx:1]: Typescripts complains about it on intelliJ, I think that's more proper way to import.

File: src/domain/landingPage/__tests__/LandingPagePreview.test.tsx (hunk 4 above)
Comment (nenonja) [reply src/domain/landingPage/__tests__/LandingPagePreview.test.tsx]: fixed

File: src/util/mockDataUtils.ts (hunk 5 above)
Comment (nenonja) [reply src/util/mockDataUtils.ts]: now fixed

File: src/util/mockDataUtils.ts (hunk 6 above)
Comment (nenonja) [reply src/util/mockDataUtils.ts]: fixed

File: src/domain/landingPage/LandingPagePreview.tsx (hunk 2 above)
Comment (Sartonon) [reply src/domain/landingPage/LandingPagePreview.tsx:1]: in tsconfig we have

`"esModuleInterop": true`

so we can import "normally"

**Review by Sartonon — APPROVED** [review]
Looks good!

**Pull request merged**